libs/Objects/src/Regex.cpp

```diff
@@ -719,6 +710,51 @@ bool Regex::derivative_with_respect_to_sym(Regex* respected_sym, const Regex* re
 	}
 }
 
+Regex* Regex::add_alt(std::vector<Regex> res, Regex* root) const {
+	if (res.size() == 1) {
+		return res[0].make_copy();
+	}
+
+	root = new Regex;
+	root->type = Type::alt;
+	root->term_l = res[0].make_copy();
+	res.erase(res.begin());
+	root->term_r = add_alt(res, Regex::cast(root->term_r, false));
+	return root;
+}
+
+Regex* Regex::to_aci(std::vector<Regex>& res) const
+{
+	// отсортировали вектор регулярок
+	std::sort(res.begin(), res.end(), [](Regex i, Regex j) {
```

Comment:
`Clang-Tidy: The parameter 'i' is copied for each invocation but only used as a const reference; consider making it a const reference `